autoPyTorch/evaluation/abstract_evaluator.py

```diff
@@ -247,13 +247,90 @@ def send_warnings_to_log(message, category, filename, lineno,
 
 
 class AbstractEvaluator(object):
+    """
+    This method defines the interface that pipeline evaluators should follow, when
+    interacting with SMAC through ExecuteTaFuncWithQueue.
+
+    An evaluator is an object that:
+        + constructs a pipeline (be it a classification or regression estimator) for a given
+          pipeline_config and run settings (budget, seed)
+        + Fits and train this pipeline (TrainEvaluator) or test a given
+          configuration (TestEvaluator)
+
+    The provided configuration determines the type of pipeline created:
```

Comment:
If we have it in the documentation in `get_pipeline` function, I think we can just give a credit to it.

Reply:
Thanks, I have improved the docstring with this!